workflows/flows/import_genome_assembly_links_flow.py

```diff
@@ -0,0 +1,385 @@
+            )
+            raise ValueError(
+                f"Missing required columns in TSV: {', '.join(missing_columns)}"
+            )
+
+        for row in reader:
+            records.append(row)
+
+    logger.info(f"Read {len(records)} records from TSV file")
+    return records
+
+
+@task(name="Process TSV records")
+def process_tsv_records(records: List[Dict[str, str]]) -> List[Dict[str, str]]:
+    """
+    Processes the TSV records, validating and cleaning the data.
+
+    Args:
+        records: List of dictionaries, one for each row in the TSV file
+
+    Returns:
+        List of validated and cleaned records
+    """
+    logger = get_run_logger()
+    validated_records = []
+
+    skipped = 0
+    for i, record in enumerate(records, 1):
+        if i == 1:
+            logger.info(f"Processing record {i}")
+            logger.info(record)
+
+        # Safely normalise fields
+        primary_assembly = (record.get("primary_assembly") or "").strip()
+        genome_raw = record.get("genome")
+        genome = (genome_raw or "").strip()
+
+        # Remove rows where genome is NA-like (None, NA, N/A, null, NULL)
+        if genome_raw is None or genome.lower() in {"na", "n/a", "null"}:
+            skipped += 1
+            continue
+
+        cleaned_record = {
+            "primary_assembly": primary_assembly,
+            "genome": genome,
+            "mag_accession": (
+                (record.get("mag_accession") or "").strip()
+                if record.get("mag_accession")
+                else None
+            ),
+            "species_rep": (
+                (record.get("species_rep") or "").strip()
+                if record.get("species_rep")
+                else None
+            ),
+        }
+
+        validated_records.append(cleaned_record)
+
+    if skipped:
+        logger.info(f"Skipped {skipped} record(s) due to NA-like genome values")
+
+    logger.info(f"Validated {len(validated_records)} records")
+    return validated_records
+
+
+@task(name="Find Genome and Assembly objects")
+def find_objects(
+    records: List[Dict[str, str]],
+) -> List[Tuple[Dict[str, str], Optional[Genome], Optional[Assembly]]]:
+    """
+    Finds the Genome and Assembly objects for each record using batched lookups
+    to reduce database round trips.
+
+    Args:
+        records: List of validated and cleaned records
+
+    Returns:
+        List of tuples containing the record, Genome object, and Assembly object
+    """
+    logger = get_run_logger()
+    results = []
+
+    if not records:
+        return results
+
+    genome_accessions = {r["genome"] for r in records if r.get("genome")}
+    assembly_accessions = {
+        r["primary_assembly"] for r in records if r.get("primary_assembly")
+    }
+
+    genomes_map = {
+        g.accession: g
+        for g in Genome.objects.filter(accession__in=list(genome_accessions)).only(
+            "genome_id", "accession"
+        )
+    }
```

Comment:
Maybe it's a good idea to count the number of genomes to make sure we ar getting the expected total?